src/extension.rs

```diff
@@ -343,6 +342,14 @@ impl ExtensionSet {
         self.0.insert(extension.clone());
     }
 
+    /// Adds a type var (which must have been declared as a [TypeParam::Extensions]) to this set
+    pub fn insert_type_var(&mut self, idx: usize) {
+        // Represent type vars as string representation of DeBruijn index.
+        // This is not a legal IdentList or ExtensionId so should not conflict.
```

Comment:
Yes, this is a hack...I tried having two HashSet's in 80d8d8d4dad110f8b7d5afea9a7e449aeff0429c but it was messier (and caused a mass of clippy errors because of making error types (that contain ExtensionSets) too big - so we'd need to add some Boxes too)